src/coreComponents/integrationTests/fluidFlowTests/testCompositionalMultiPhaseMFDPolyhedral.cpp

```diff
@@ -0,0 +1,722 @@
+
+protected:
+  void SetUp() override
+  {
+    testBinaryDir = TEST_BINARY_DIR;
+    std::string meshFile = testBinaryDir + std::string( "/" ) + GetParam();
+    std::string xmlInput = generateXmlInputCompMFD( INNER_TPFA, meshFile );
+    setupProblemFromXML( state.getProblemManager(), xmlInput.c_str() );
+  }
+
+  GeosxState state;
+  std::string testBinaryDir;
+};
+
+INSTANTIATE_TEST_SUITE_P(
+  MeshFiles,
+  CompositionalMFDTPFAIntegrationTest,
+  ::testing::Values(
+    "polyhedral_voronoi_regular.vtu",
+    "hex_pyr_tet_nested_mixed.vtu",
+    "polyhedral_voronoi_lattice.vtu",
+    "polyhedral_voronoi_complex.vtu"
+    )
+  );
+
+TEST_P( CompositionalMFDTPFAIntegrationTest, PressureFieldL2Error )
+{
+  ProblemManager & problemManager = state.getProblemManager();
+  DomainPartition & domain = problemManager.getDomainPartition();
+
+  auto & solver =
+    dynamic_cast< CompositionalMultiphaseHybridFVM & >(
+      problemManager.getPhysicsSolverManager().getGroup< CompositionalMultiphaseHybridFVM >( "FlowSolverMFD" ) );
+
+  // One implicit step with dt = 0.01
+  solver.setupSystem( domain, solver.getDofManager(),
+                      solver.getLocalMatrix(), solver.getSystemRhs(),
+                      solver.getSystemSolution() );
+  solver.implicitStepSetup( 0.0, TIME_STEP, domain );
+  solver.solverStep( 0.0, TIME_STEP, 0, domain );
+  solver.updateState( domain );
+  solver.implicitStepComplete( 0.0, TIME_STEP, domain );
+
+  MeshLevel & mesh = domain.getMeshBody( 0 ).getBaseDiscretization();
+  CellElementSubRegion & subRegion = mesh.getElemManager().getRegion( 0 ).getSubRegion< CellElementSubRegion >( 0 );
+
+  arrayView2d< real64 const > centers = subRegion.getElementCenter();
+  arrayView1d< real64 const > volumes = subRegion.getElementVolume();
+  arrayView1d< real64 const > const p_h = subRegion.getField< fields::flow::pressure >();
+
+  RAJA::ReduceSum< parallelDeviceReduce, real64 > l2Error_ReduceSum( 0.0 );
+  RAJA::ReduceSum< parallelDeviceReduce, real64 > totalVolume_ReduceSum( 0.0 );
+  localIndex const n_cells  = subRegion.size();
+  forAll< geos::parallelDevicePolicy<> >( n_cells, [=] GEOS_HOST_DEVICE ( localIndex const i )
+  {
+    real64 x = centers[i][0];
+    real64 volume = volumes[i];
+    real64 pNumeric = p_h[i];
+    real64 pExact = 2.0 * (1.0 - x) + 1.0 * x;
+    l2Error_ReduceSum += (pNumeric - pExact) * (pNumeric - pExact) * volume;
+    totalVolume_ReduceSum += volume;
+  } );
+
+  real64 const data[2] = { l2Error_ReduceSum.get(), totalVolume_ReduceSum.get() };
+  real64 l2Error = std::sqrt( data[0] ) / data[1];
+
+  // Expect exact solution only on the k-orthogonal regular mesh; expect non-zero error on mixed mesh
+  std::string meshFile = GetParam();
+  if( meshFile == std::string( "polyhedral_voronoi_regular.vtu" ) )
+  {
+    EXPECT_NEAR( l2Error, 0.0, PRESSURE_L2_TOLERANCE );
+  }
+  else
+  {
+    EXPECT_GT( l2Error, PRESSURE_L2_TOLERANCE );
+  }
+}
```

Comment:
See [suggestion below](https://github.com/GEOS-DEV/GEOS/pull/3765/files#r2482954535) for `CompositionalMFDNonTPFAExactnessTest`

Reply:
Thank you, Nicola. Your suggestion was turned into a helper function and is now called when appropriate.